File: hw/pilatasig/u2/HW_27_UnitTest/ec/edu/espe/controller/BasicOperationController.js
class BasicOperationsController{
    BasicOperationsController(basicOperations){
        this.basicOperations= basicOperations;
    }

    add(term1,term2){
        let sum;
        sum=term1+term2;
        return sum;
    }

    add(){
        let sum;
        sum=this.basicOperations.getTerm1()+this.basicOperations.getTerm2();
        return sum;
    }
}

File: ws/robalino/u2/Ws24UnitTest/Calculator/src/main/java/ec/edu/espe/calculator/controller/BasicOperationController.java


package ec.edu.espe.calculator.controller;

import ec.edu.espe.calculator.model.BasicOperation;

/**
 *
 * @author Benjamin Robalino <jabasteam>
 */

public class BasicOperationController {
    private BasicOperation basicOperation;
    private float add(BasicOperation basicOperation){
        float sum ;
        sum = basicOperation.getTerm1() + basicOperation.getTerm2();
        return sum;
    }

    public BasicOperationController(BasicOperation basicOperation) {
        this.basicOperation = basicOperation;
    }
    
    
    public float add(float addend1, float addend2){
        float sum;
        sum = addend1+addend2;
        return sum;
        
    }
    

    public BasicOperationController() {
        this.basicOperation = basicOperation;
    }

    /**
     * @return the basicOperation
     */
    public BasicOperation getBasicOperation() {
        return basicOperation;
    }

    /**
     * @param basicOperation the basicOperation to set
     */
    public void setBasicOperation(BasicOperation basicOperation) {
        this.basicOperation = basicOperation;
    }
    
}
